web/src/user_profile.ts

```diff
@@ -616,15 +612,18 @@ function add_user_to_groups(group_ids: number[], user_id: number, $alert_box: JQ
     add_user_to_next_group();
 }
 
-export function show_user_profile(user: User, default_tab_key = "profile-tab"): void {
+export async function show_user_profile(
+    user: User,
+    default_tab_key = "profile-tab",
+): Promise<void> {
     const field_types = realm.custom_profile_field_types;
     const profile_data = realm.custom_profile_fields
         .flatMap((f) => get_custom_profile_field_data(user, f, field_types) ?? [])
         .filter((f) => f.name !== undefined);
     original_values = {
         user_id: user.user_id.toString(),
     };
-    const user_unsub_streams = get_user_unsub_streams(user.user_id);
+    const user_unsub_streams = await get_user_unsub_streams(user.user_id);
```

Comment:
While it's reasonable to call `get_user_unsub_streams` here to launch the fetch (if needed) preemptively, I don't think we should `await` on it unless we're in the "streams" tab of the relevant view? What you have delays displaying the profile data even though nothing on the default tab depends on the fetch.

Reply:
`user_unsub_streams` is used to calculate `show_user_subscribe_widget` in the next line of code, which seems like it almost always will run. Do you want to change that logic?

Reply:
As discussed there, opened [#36976](https://github.com/zulip/zulip/pull/36976)